v2/routes.go

```diff
@@ -52,50 +50,125 @@ import (
 //	204: noContent
 func (s *Server) createSubscription(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
-	var response *http.Response
 	bodyBytes, err := io.ReadAll(r.Body)
 	if err != nil {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
 	sub := pubsub.PubSub{}
 	if err = json.Unmarshal(bodyBytes, &sub); err != nil {
-		respondWithError(w, "marshalling error")
+		respondWithStatusCode(w, http.StatusBadRequest, fmt.Sprintf("marshalling error %v", err))
 		localmetrics.UpdateSubscriptionCount(localmetrics.FAILCREATE, 1)
 		return
 	}
-	if sub.GetEndpointURI() != "" {
-		response, err = s.HTTPClient.Post(sub.GetEndpointURI(), cloudevents.ApplicationJSON, nil)
-		if err != nil {
-			log.Printf("there was error validating endpointurl %v, subscription wont be created", err)
-			localmetrics.UpdateSubscriptionCount(localmetrics.FAILCREATE, 1)
-			respondWithError(w, err.Error())
-			return
-		}
-		defer response.Body.Close()
-		if response.StatusCode != http.StatusNoContent {
-			log.Printf("there was an error validating endpointurl %s returned status code %d", sub.GetEndpointURI(), response.StatusCode)
-			respondWithError(w, "return url validation check failed for create subscription.check endpointURI")
-			localmetrics.UpdateSubscriptionCount(localmetrics.FAILCREATE, 1)
-			return
-		}
+	endPointURI := sub.GetEndpointURI()
+	if endPointURI == "" {
+		respondWithStatusCode(w, http.StatusBadRequest, "EndpointURI can not be empty")
+		localmetrics.UpdateSubscriptionCount(localmetrics.FAILCREATE, 1)
+		return
 	}
-	// check sub.EndpointURI by get
-	sub.SetID(uuid.New().String())
-	_ = sub.SetURILocation(fmt.Sprintf("http://localhost:%d%s%s/%s", s.port, s.apiPath, "subscriptions", sub.ID)) //nolint:errcheck
+	if subExists, ok := s.pubSubAPI.HasSubscription(sub.GetResource()); ok {
+		respondWithStatusCode(w, http.StatusConflict,
+			fmt.Sprintf("subscription (id: %s) with same resource already exists, skipping creation",
+				subExists.GetID()))
+		localmetrics.UpdateSubscriptionCount(localmetrics.FAILCREATE, 1)
+		return
+	}
+
+	id := uuid.New().String()
+	sub.SetID(id)
+	sub.SetVersion(API_VERSION)
+	sub.SetURILocation(fmt.Sprintf("http://localhost:%d%s%s/%s", s.port, s.apiPath, "subscriptions", sub.ID)) //nolint:errcheck
 
 	newSub, err := s.pubSubAPI.CreateSubscription(sub)
```

Comment:
don't need local pubsub anymore, since we are using configmap

Reply:
delete this

Reply:
added note, will do in the next story